engine/access/handle_irrecoverable_state_test.go

```diff
@@ -0,0 +1,259 @@
+package access
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/antihax/optional"
+	accessproto "github.com/onflow/flow/protobuf/go/flow/access"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	restclient "github.com/onflow/flow/openapi/go-client-generated"
+
+	"github.com/onflow/flow-go/crypto"
+	accessmock "github.com/onflow/flow-go/engine/access/mock"
+	"github.com/onflow/flow-go/engine/access/rest"
+	"github.com/onflow/flow-go/engine/access/rest/routes"
+	"github.com/onflow/flow-go/engine/access/rpc"
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	statestreambackend "github.com/onflow/flow-go/engine/access/state_stream/backend"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/grpcserver"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/metrics"
+	module "github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/network"
+	protocol "github.com/onflow/flow-go/state/protocol/mock"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/grpcutils"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// IrrecoverableStateTestSuite tests that Access node indicate an inconsistent or corrupted node state
+type IrrecoverableStateTestSuite struct {
+	suite.Suite
+	state      *protocol.State
+	snapshot   *protocol.Snapshot
+	epochQuery *protocol.EpochQuery
+	log        zerolog.Logger
+	net        *network.EngineRegistry
+	request    *module.Requester
+	collClient *accessmock.AccessAPIClient
+	execClient *accessmock.ExecutionAPIClient
+	me         *module.Local
+	chainID    flow.ChainID
+	metrics    *metrics.NoopCollector
+	rpcEng     *rpc.Engine
+	publicKey  crypto.PublicKey
+
+	// storage
+	blocks       *storagemock.Blocks
+	headers      *storagemock.Headers
+	collections  *storagemock.Collections
+	transactions *storagemock.Transactions
+	receipts     *storagemock.ExecutionReceipts
+
+	ctx irrecoverable.SignalerContext
+
+	// grpc servers
+	secureGrpcServer   *grpcserver.GrpcServer
+	unsecureGrpcServer *grpcserver.GrpcServer
+}
+
+func (suite *IrrecoverableStateTestSuite) SetupTest() {
```

Comment:
Since it's a new test suite we shouldn't use `new(mocked_interface)` anymore instead use `NewMockedInterface(suite.T())` that is more restrictive assertions wise.